java/lib/core/src/main/java/org/eclipse/tahu/message/model/SparkplugBPayloadDataMap.java

```diff
@@ -0,0 +1,344 @@
+/*
+ * Licensed Materials - Property of Cirrus Link Solutions
+ * Copyright (c) 2024 Cirrus Link Solutions LLC - All Rights Reserved
+ * Unauthorized copying of this file, via any medium is strictly prohibited
+ * Proprietary and confidential
+ */
+package org.eclipse.tahu.message.model;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+import org.eclipse.tahu.message.model.SparkplugBPayloadMap.SparkplugBPayloadMapBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+/**
+ * A class representing a Sparkplug B payload as a {@link Map} to prevent duplication of {@link Metric}s. This can be
+ * useful for Sparkplug BIRTH payloads
+ */
+public class SparkplugBPayloadDataMap extends SparkplugBPayload {
+
+	private static Logger logger = LoggerFactory.getLogger(SparkplugBPayloadMap.class.getName());
```

Comment:
I think this should be 'logger = LoggerFactory.getLogger(SparkplugBPayloadDataMap.class.getName());'

Reply:
Thanks - fixed